### PR TITLE
add try/exception to handle UnicodeDecodeError

### PR DESCRIPTION
**Pull request by aashishc1988**
Due to lack of pull-up on different parts, the re-initialization of uart causes it to send an invalid character which we are not able to handle properly on htrun side. While partners are investigating the issue and fix, we will add this change in to handle the exception gracefully, as this bug is blocking the addition of multiple parts to our CI

### CODE REVIEW

**Comment by aashishc1988** [comment]
@cmonr Added a print statement in case unicodeDecode error occurs.

File: mbed_host_tests/host_tests_conn_proxy/conn_proxy.py [hunk 1]
```diff
@@ -42,7 +42,11 @@ def __init__(self):
 
     def append(self, payload):
         """! Append stream buffer with payload and process. Returns non-KV strings"""
-        self.buff += payload.decode('utf-8')
+        logger = HtrunLogger('CONN')
+        try:
+            self.buff += payload.decode('utf-8')
+        except UnicodeDecodeError:
+            logger.prn_err("UnicodeDecodeError encountered!")
```
Comment (bridadan) [inline mbed_host_tests/host_tests_conn_proxy/conn_proxy.py]: Total nit pick here: I'd go with a `prn_wrn` here since the `htrun` will continue to run.

Comment (aashishc1988) [reply mbed_host_tests/host_tests_conn_proxy/conn_proxy.py]: logger.prn_err seemed like an obvious choice at first, but then looking at other places where it is used, we do return/raise an exception when we log the message with prn_err, but here we won't be doing anything like that, so I see your point. I will change that.

**Comment by cmonr** [comment]
@aashishc1988 Something I was wondering.

Since the issue is known to be target specific, and a patch for the target appears to be feasable, why not open an Mbed OS PR instead? I'd like to echo @theotherjimmy sentiment about temporary, target-specific tool patches (https://github.com/ARMmbed/greentea/pull/278#issuecomment-407175402) in that I think we'd be much better off if we can avoid them.

**Comment by bridadan** [comment]
For what its worth, I think htrun should be able to handle these unicode errors, even if a patch lands in the target. If you want a patch to land in Mbed OS too then no worries, but I don't mind htrun being cautious here as well.

**Comment by studavekar** [comment]
I agree with @bridadan test runner should handle Unicode character and should not crash. It effectively fixes htrun issue https://github.com/ARMmbed/htrun/issues/206.  

There is an issue on mbed-os https://github.com/ARMmbed/mbed-os/issues/7511 for ST device, we can add other targets which show this behaviour.

**Comment by theotherjimmy** [comment]
how about decode with ignore or decode with replacement? that way we only drop/replace the offending char.

**Comment by aashishc1988** [comment]
@theotherjimmy using decode with ignore/replace will mask the error, I am not entirely sure what we plan on doing with this sort of behavior in future but it would be good to have some sort of message printed out if/when we encounter this issue. What if tomorrow we add a new part which has the same issue, but now since we are masking it, we wouldnt be able to know and push the partners for the fix.

Also, regarding the solution to treat it as bytes object, wouldnt that require change on both greentea and htrun? If you thinkthat is the most apt solution, lets take this offline and discuss what all need to change and the effects of it. I am still fairly new to GT and htrun architecture :)

**Comment by theotherjimmy** [comment]
@aashishc1988 Could you have the except do a decode with ignore? that way we minimize the lost bytes.

**Comment by aashishc1988** [comment]
Yes, thats a good point. We can print a UnicodeDecodeError exception message, and "decode with ignore" as well. Made the change.

**Comment by theotherjimmy** [comment]
@cmor care to review?

**Comment by cmonr** [comment]
I like it! Catches the error, indicates that one happened, and still tries to return the data.

Commit message could be shorter/multiline, but I'm not going to block on a nit.

**Comment by jeromecoutant** [comment]
Hi
Is it possible to release this patch in a version ?
Thx

**Comment by cmonr** [comment]
@jeromecoutant We'll likely be making another release sometime mid next month (if not sooner) with the release of Mbed OS 5.10. There are one or two fixes looking to come in that will need multiple tools to be updated in lock step.

**Comment by dhwalters423** [comment]
@MarceloSalazar this is blocking support for the UBLOX ODIN W2 and SMCC tests.

**Comment by aashishc1988** [comment]
@dhwalters423 Just fyi, this is not target specific. We have seen it on platforms other than ST (I cannot recall names, would need help of a colleague who is out today). 
Also, it looks like the current PR is merged. Are you still seeing an issue?

**Comment by dhwalters423** [comment]
When installing newest htrun from sources I do not see the issue anymore. thanks for clarification.

**Pull request merged**